docker-entrypoint.sh

```diff
@@ -1,61 +1,366 @@
+mkdir -p "${ORS_HOME}"/{files,logs,graphs,elevation_cache,config} || warning "Could not create ${ORS_HOME} and folders"
+debug "Populated ORS_HOME=${ORS_HOME} with the default folders: files, logs, graphs, elevation_cache"
 
+# Check if the original jar file exists
+if [ ! -f "${jar_file}" ]; then
+  critical "Jar file not found. This shouldn't happen. Exiting."
+fi
 
-if [ -z "${CATALINA_OPTS}" ]; then
-  export CATALINA_OPTS="-Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.port=9001 -Dcom.sun.management.jmxremote.rmi.port=9001 -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -Djava.rmi.server.hostname=localhost"
+
+# Check that ors_engine_graphs_root_path is not empty and not set to /
+if [ -n "${ors_engine_graphs_root_path}" ] && [ "${ors_engine_graphs_root_path}" = "/" ]; then
+  critical "ors.engine.graphs_root_path is set to /. This is not allowed. Exiting."
+else
+  debug "ors.engine.graphs_root_path=${ors_engine_graphs_root_path} is set and not empty and not set to /"
 fi
 
-if [ -z "${JAVA_OPTS}" ]; then
-  export JAVA_OPTS="-Djava.awt.headless=true -server -XX:TargetSurvivorRatio=75 -XX:SurvivorRatio=64 -XX:MaxTenuringThreshold=3 -XX:+UseG1GC -XX:+ScavengeBeforeFullGC -XX:ParallelGCThreads=4 -Xms1g -Xmx2g"
+# Check that ors_engine_elevation_cache_path is not empty and not set to /
+if [ -n "${ors_engine_elevation_cache_path}" ] && [ "${ors_engine_elevation_cache_path}" = "/" ]; then
+  critical "ors.engine.elevation.cache_path is set to /. This is not allowed. Exiting."
+else
+  debug "ors.engine.elevation.cache_path=${ors_engine_elevation_cache_path} is set and not empty and not set to /"
 fi
 
-{
-  echo "CATALINA_BASE=\"${catalina_base}\""
-  echo "CATALINA_HOME=\"${catalina_base}\""
-  echo "CATALINA_PID=\"${catalina_base}/temp/tomcat.pid\""
-  echo "CATALINA_OPTS=\"${CATALINA_OPTS}\""
-  echo "JAVA_OPTS=\"${JAVA_OPTS}\""
-} >"${catalina_base}"/bin/setenv.sh
+# Update the example-ors-config.env and example-ors-config.yml files if they don't exist or have changed
+update_file "${ORS_HOME}/config/example-ors-config.env" "/example-ors-config.env" "true"
+update_file "${ORS_HOME}/config/example-ors-config.yml" "/example-ors-config.yml" "true"
 
-if [ "${BUILD_GRAPHS}" = "True" ]; then
-  rm -rf "${graphs:?}"/*
+# The config situation is difficult due to the recent ors versions.
+# To ensure a smooth transition, we need to check if the user is using a .json file or a .yml file.
+# If neither is set, we need to print a migration info and default to the example-ors-config.env file.
+# Check if ors_config_location is a .json file and exists
+if [[ "${ors_config_location}" = *.yml ]] && [[ -f "${ors_config_location}" ]]; then
+  success "Using yml config: ${ors_config_location}"
+elif [[ "${ors_config_location}" = *.json ]] && [[ -f "${ors_config_location}" ]]; then
+  success "Using json config: ${ors_config_location}"
```

Comment:
Here, we could note that we're using the values from the ENV?

Reply:
yep

Reply:
I added the following line after this if else construct
```
info "Any ENV variables will have precedence over configuration variables from config files."
```

Reply:
That is valid information for the user, but these two lines are only printed if the value is read from an ENV. So I'm thinking along the lines of "Using yml config from ENV" or similar?

Reply:
I'll change it to the following:
```
success "Using yml config from ENV: ${ors_config_location}"
success "Using json config from ENV: ${ors_config_location}"
```